### DISCUSSION TITLE
How do I use date fields in postgres as a string?

### DISCUSSION THREAD

**Discussion by AlexJWayne** [prisma/prisma#16973] [Q&A]
In TypeORM we had:

```typescript
  @Column('date')
  date: string
```

This stored dates without times, as strings. Which is handy because when you dont have time, you also don't have a time _zone_.

But it seems Prisma works differently.

```
  date          DateTime        @db.Date
```

This generates the following typescript prop:

```typescript
  date: Date
```

So now to add a date from our JSON api, we have find it, parse it as a date, and then _hope_ we got the timezone correct or it will be a day off from what was intended.

I'm really hoping there is someway to:

```typescript
prisma.myModel.create({
  data: { date: '2022-12-21' }
})
```

Instead of like:

```typescript
prisma.myModel.create({
  data: { date: new Date(Date.UTC(2022, 11, 21)) }
})
```

Or do I just drop the column type down to `string` and rely on my app code to validate the format here?

---

Furthermore, if I pass that UTC date to a record, and then fetch it again prisma gives me back a date object that is equivalent to:

```
2022-12-21T00:00:00.000Z
```

This seems incorrect. In the database it's just a date, but to assume that it's UTC midnight on that date makes a lot of assumptions.

* It might mean a whole 24 hour period, like if it rained at all that day.
* It might be a start date in which case midnight makes sense.
* It might be an end date in which case midnight of the following date makes more sense.
* It might be a due date, which means the end of that day makes sense

Now let's assume that this date is related to place that does have a specific time zone. And I want get two dates for the start and end of that day. In order to use the string `2022-01-01` for that I have to interpret that with a time zone. To not do so is a type error. But if it's already a `Date`, then when I do that I now have 4pm - 4pm (assuming pacific time) instead of midnight to midnight because the server assumed UTC for a date with no time at all.

---

Lastly, this seems maddening:

```typescript
const model = await prisma.myModel.create({
  data: { date: new Date('2022-12-21T23:51:44.766Z') }
})
console.log(model.date) // 2022-12-21T00:00:00.000Z
```

Accepting a time, then throwing it out and returning a made up time seems like a really bad idea.

**Comment by nurul3101** [comment]
Hello @AlexJWayne 👋 

Thanks for sharing this feedback, we have a similar related GitHub Issue which discusses the same:
- https://github.com/prisma/prisma/issues/4355

I have shared this internally with the team so that we can investigate the possibility of prioritising this.
Meanwhile, does this approach work for you?
```typescript
prisma.myModel.create({
  data: { date: new Date('2022-12-21') }
})
```

**Comment by drewlowe** [comment]
Here's a very low overhead solution using date-fns and date-fns-tz:
```js
const date = toZonedTime(new Date(dateFromPrisma)), "UTC")
const formattedDate = format((date), 'MM/dd/yyyy')
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
